test/src/test/java/org/corfudb/integration/LogReplicationReaderWriterIT.java

```diff
@@ -354,9 +355,40 @@ public static void writeLogEntryMsgs(List<LogReplicationEntryMsg> msgQ, Set<Stri
             log.debug("msgQ is EMPTY");
         }
 
-        for (LogReplicationEntryMsg msg : msgQ) {
+        Iterator<LogReplicationEntryMsg> msgQIterator = msgQ.iterator();
```

Comment:
I think we should have an E2E test, where we actually intersect the ACK and drop it. We don't ACK on a 1:1 basis but we have aggregated ACKs so I'm wondering if this really replicates the scenario we're trying to target.

Reply:
I realized while working on the E2E tests that we do ACK 1:1 basis once we process the msg.
 ```
if (sinkManager.processMessage(dataMessage)) {
                ackCnt++;
                ....
}
...
if (shouldAck()) {
     // send ACK
}


public boolean shouldAck() {
        long currentTime = java.lang.System.currentTimeMillis();
        ackCnt++;

        if (ackCnt >= ackCycleCnt || (currentTime - ackTime) >= ackCycleTime) {
           .....
            return true;
        }
       return false
}
```
 And "ackCycleCnt" value is 2